File: modules/api/src/main/java/com/natevaughan/kchat/api/MessageRepo.java
package com.natevaughan.kchat.api;

import java.time.Instant;

/**
 * Created by nate on 3/8/19
 */
public interface MessageRepo extends Repository<Message> {
	Iterable<Message> findForChatSinceTimestamp(Chat chat, Instant timestamp);
	Iterable<Message> findRecent(Chat chat, Integer count);
}

File: modules/chat-api/src/main/kotlin/com/natevaughan/kchat/api/MessageRepo.kt
package com.natevaughan.kchat.api

import java.time.Instant

/**
 * Created by nate on 3/8/19
 */
interface MessageRepo : Repository<Message> {
	fun findForChatSinceTimestamp(chat: Chat, timestamp: Instant): Iterable<Message>
	fun findRecent(chat: Chat, count: Int): Iterable<Message>
}
